### PR TITLE
allow reviewers to comment through pull request mode

### PR DESCRIPTION
**Pull request by benoitf**

### CODE REVIEW

File: doc/Authoring_Plugins.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Let's create our first Theia plug-in. As an example, we are going to register a command _Hello World_ that displays a notification "Hello world!". This article is guiding you through all the necessary steps.
+
+## Theia’s Architecture
+
+
+### Plug-in vs Extension
+
+Another way of extending Theia is using Theia extensions. Here are the main differences between plug-ins vs the extensions.
+
+pros:
```
Comment (sunix) [inline doc/Authoring_Plugins.md]: not sure which plugin-in or extension these are pros or cons. Maybe we should emphasis the fact that we are introducing a new extension mechanism within Theia and that there is new terminologies: Plugin AND extension. Talk a bit about what is an extension + pros and con as well ?

File: doc/Authoring_Plugins.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+In Theia, a plug-in has access to the API through an object named `theia` which is available in all plug-ins.
+[More details on API]](https://github.com/theia-ide/theia/blob/master/packages/plugin/API.md).
+
+There are two natures of plug-ins:
+ - Backend plug-in. If you're familiar with VS Code extensions, it's very close. The plug-in's code is running in its own process on the server side. API is called and it's the API that will send some actions on user's browser/UI to register new commands, etc. All the callbacks are executed on the server side on a dedicated process.
+ - Frontend plug-in. In that case, callbacks are executed in a worker thread on the UI/browser. These plug-ins are only authorized to use "browser compliant" modules. For example opening or writing to a file is impossible as all the code of the plug-in is running on the browser side. But this approach is helpful if you really want to have some stuff on the client side to avoid some network operations.
+
+## Prerequisites
+
+Having a running Theia instance. (v0.3.12+)
```
Comment (sunix) [inline doc/Authoring_Plugins.md:43]: why not proposing  as an alternative, an additional factory url ? AND/OR point to the theia installation page. Any theia instance would work or do we need mandatory extension so this is working ?

File: doc/Authoring_Plugins.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      "publisher": "theia",
+      "keywords": [
+        "theia-plugin"
+      ],
+      "version": "0.0.1",
+      "files": [
+        "src"
+      ],
+      "devDependencies": {
+        "@theia/plugin": "latest",
```
Comment (sunix) [inline doc/Authoring_Plugins.md]: version latest is confusing, maybe provide a link to the npm page to help

**Review by sunix — CHANGES_REQUESTED** [review]

File: doc/Authoring_Plugins.md (hunk 2 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md:43]: Factory URL will probably come within Che documentation
ok for adding like on how to grab theia
any theia instance

File: doc/Authoring_Plugins.md (hunk 3 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md]: ok

File: doc/Authoring_Plugins.md (hunk 1 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md]: I've rewritten at https://hackmd.io/qqNqVr5KT32aUhNTypYXJg?both

File: doc/Authoring_Plugins.md (hunk 2 above)
Comment (benoitf) [reply doc/Authoring_Plugins.md:43]: about factory I think we will add link when it'll done

**Pull request closed without merging**